lib/v2/components/search_user/search_user_widget.dart

```diff
@@ -70,17 +71,17 @@ class SearchUserWidget extends StatelessWidget {
       case PageState.initial:
         return const SizedBox.shrink();
       case PageState.loading:
-        return SearchUsersList(users: state.users);
+        return SearchUsersList(users: state.users, resultCallBack: resultCallBack);
       case PageState.failure:
-        return SearchUsersList(users: state.users);
+        return SearchUsersList(users: state.users, resultCallBack: resultCallBack);
```

Comment:
You could have avoided passing parameters to `SearchUsersList` and `SearchResultRow` by making use of a `Blocbuilder` within them.

Blocbuilder not only **rebuilds** it also gives you access to the **state** of the bloc you use.

Also `resultCallBack` would fit well inside a `BlocListener` you would not have to pass it as a parameter in each declaration of `SearchUsersLists` you only execute an event and change a state listen to the listener and it executes the callback.

Reply:
moved to use bloc builder. 
part of the follow up pr